scipy/linalg/_solvers.py

```diff
@@ -304,12 +310,20 @@ def solve_discrete_are(a, b, q, r):
     .. math::
         X = A'XA-(A'XB)(R+B'XB)^-1(B'XA)+Q
 
-    It is solved directly using a Schur decomposition method.
+    It is solved via forming the extended symplectic pencil    
+    .. math:`H - \lambda J` given by the block matrices
+        
+    
+           [  A   0   B ]             [ I   0   B ] 
```

Comment:
Should this just define `H = [...]`, `J = [...]` instead of restating the equation? I guess you could also take out the first `H - \lambda J`.